From be296aee0648c279d83e3eaf09835684c6e67731 Mon Sep 17 00:00:00 2001
From: Jaswant Panchumarti <jaswant.panchumarti@kitware.com>
Date: Wed, 29 May 2024 19:33:08 -0400
Subject: [PATCH] Install JS library files in vtk.wasm

- closes https://gitlab.kitware.com/vtk/vtk/-/issues/19356
- Provides the .js files in both build and install interface.
- This fixes the use case where someone links to an installed VTK,
rather than doing so from the buildsystem.
---
 Rendering/UI/CMakeLists.txt | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Rendering/UI/CMakeLists.txt b/Rendering/UI/CMakeLists.txt
index e18a1639b61..9e132569198 100644
--- a/Rendering/UI/CMakeLists.txt
+++ b/Rendering/UI/CMakeLists.txt
@@ -114,7 +114,10 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
   else ()
     vtk_module_link_options(VTK::RenderingUI
       INTERFACE
-        "--js-library=${CMAKE_CURRENT_SOURCE_DIR}/vtkWebAssemblyRenderWindowInteractor.js")
+        "$<BUILD_INTERFACE:--js-library=${CMAKE_CURRENT_SOURCE_DIR}/vtkWebAssemblyRenderWindowInteractor.js>"
+        "$<INSTALL_INTERFACE:--js-library=$<INSTALL_PREFIX>/lib/vtkWebAssemblyRenderWindowInteractor.js>")
+    install(FILES "${CMAKE_CURRENT_SOURCE_DIR}/vtkWebAssemblyRenderWindowInteractor.js"
+      DESTINATION ${CMAKE_INSTALL_LIBDIR})
   endif ()
 endif ()
 
-- 
GitLab

